Source/WebCore/bindings/js/JSFileSystemWritableFileStreamCustom.cpp

```diff
@@ -0,0 +1,100 @@
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "JSFileSystemWritableFileStream.h"
+
+#include "InternalWritableStream.h"
+#include "JSDOMPromiseDeferred.h"
+#include "WebCoreJSClientData.h"
+#include <JavaScriptCore/CallFrame.h>
+
+namespace WebCore {
+
+static JSC::JSValue writeImpl(FileSystemWritableFileStream& stream, JSC::JSGlobalObject& lexicalGlobalObject, JSC::JSValue data)
+{
+    auto& vm = JSC::getVM(&lexicalGlobalObject);
+    auto* clientData = static_cast<JSVMClientData*>(vm.clientData);
+
+    JSC::MarkedArgumentBuffer arguments;
+    arguments.append(stream.internalWritableStream().guardedObject());
+    ASSERT(!arguments.hasOverflowed());
+    auto& writerPrivateName = clientData->builtinFunctions().writableStreamInternalsBuiltins().acquireWritableStreamDefaultWriterPrivateName();
+    auto writerOrException = InternalWritableStream::invokeWritableStreamFunction(lexicalGlobalObject, writerPrivateName, arguments);
+    if (writerOrException.hasException())
+        return { };
+
+    auto writer = writerOrException.releaseReturnValue();
+    arguments.clear();
+    arguments.append(writer);
+    arguments.append(data);
+    ASSERT(!arguments.hasOverflowed());
+    auto& writePrivateName = clientData->builtinFunctions().writableStreamInternalsBuiltins().writableStreamDefaultWriterWritePrivateName();
+    auto result = InternalWritableStream::invokeWritableStreamFunction(lexicalGlobalObject, writePrivateName, arguments);
+
+    arguments.clear();
+    arguments.append(writer);
+    auto& releasePrivateName = clientData->builtinFunctions().writableStreamInternalsBuiltins().writableStreamDefaultWriterReleasePrivateName();
+    InternalWritableStream::invokeWritableStreamFunction(lexicalGlobalObject, releasePrivateName, arguments);
+    if (result.hasException())
+        return { };
+
+    return result.releaseReturnValue();
+}
+
+JSC::JSValue JSFileSystemWritableFileStream::write(JSC::JSGlobalObject& lexicalGlobalObject, JSC::CallFrame& callFrame)
+{
+    return writeImpl(wrapped(), lexicalGlobalObject, callFrame.argument(0));
+}
+
+JSC::JSValue JSFileSystemWritableFileStream::seek(JSC::JSGlobalObject& lexicalGlobalObject, JSC::CallFrame& callFrame)
+{
+    std::optional<uint64_t> position;
+    {
+        auto scope = DECLARE_THROW_SCOPE(lexicalGlobalObject.vm());
+        auto positionConversionResult = convert<IDLUnsignedLong>(lexicalGlobalObject, callFrame.argument(0));
+        if (!positionConversionResult.hasException(scope))
+            position = positionConversionResult.returnValue();
+    }
+    FileSystemWritableFileStream::WriteParams params { FileSystemWriteCommandType::Seek, std::nullopt, position, std::nullopt };
+    auto data = toJS<IDLDictionary<FileSystemWritableFileStream::WriteParams>>(lexicalGlobalObject, *JSC::jsCast<JSDOMGlobalObject*>(&lexicalGlobalObject), params);
```

Comment:
toJS may trigger an exception in workers for termination.
We should probably either check this or make writeImpl check for an existing exception.

Reply:
I made data undefined if there is an exception. 
I didn't return early as the [spec](https://fs.spec.whatwg.org/#api-filesystemwritablefilestream-seek) says return the result of "writing a chunk", so I guess we want to invoke `writableStreamDefaultWriterWrite` function (and `FileSystemWritableFileStreamSink::write`) with an invalid parameter.
Let me know if you think this should be done differently.